gen/field_test.go

```diff
@@ -76,3 +78,54 @@ func TestFieldLabelConflict(t *testing.T) {
 		})
 	}
 }
+
+func TestCompileJSONTag(t *testing.T) {
+	tests := []struct {
+		desc             string
+		required         bool
+		options          []string
+		isOmitemptyAdded bool
+	}{
+		{
+			desc:             "for optional fields",
+			required:         false,
+			options:          make([]string, 0),
+			isOmitemptyAdded: true,
+		}, {
+			desc:             "for required fields",
+			required:         true,
+			options:          make([]string, 0),
+			isOmitemptyAdded: false,
+		}, {
+			desc:             "when !omitempty tag is present",
+			required:         false,
+			options:          []string{notOmitempty},
+			isOmitemptyAdded: false,
+		},
+	}
+
+	fieldName := "numbers"
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			fieldSpec := &compile.FieldSpec{
+				ID:   0,
+				Name: fieldName,
+				Type: &compile.ListSpec{
+					ValueSpec: &compile.I64Spec{},
+				},
+				Required: tt.required,
+				Annotations: compile.Annotations{
+					goTagKey: fmt.Sprintf(
+						"json:%s,%s",
+						fieldName,
+						strings.Join(tt.options, ","),
```

Comment:
The part after the `json:` should be quoted so the result is `json:"foo,bar"`.
Also, can we build this value in a local variable above? The fmt.Sprintf and strings.Join is getting pretty loud here.